tensorboard/webapp/widgets/card_fob/card_fob_controller_component.ts

```diff
@@ -123,10 +128,48 @@ export class CardFobControllerComponent {
     return this.axisDirection === AxisDirection.VERTICAL;
   }
 
+  private shouldSwapFobs(newStep: number) {
+    return this.currentDraggingFob === Fob.END
```

Comment:
IMO this does not read well. I would rather have 

(this.currentDraggingFob === Fob.END && newStep < this.timeSelection.start.step) || (this.currentDraggingFob === Fob.END && newStep > this.timeSelection.end?.step)

newStep > undefined will always return false so I do not think you need to check this.timeSelection.end

Reply:
That way works for me.